x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlParser.java

```diff
@@ -172,5 +185,20 @@ public void setTokenFactory(TokenFactory<?> factory) {
         public TokenFactory<?> getTokenFactory() {
             return delegate.getTokenFactory();
         }
+
+        private void checkParamType(boolean param1, boolean param2) {
+            if (param1 || param2) {
+                throw new ParsingException(
+                    "Mixed parameter types is not supported in the query [AnonymousParam="
+                        + paramType[0]
+                        + ", PositionalParam="
+                        + paramType[1]
+                        + ", NamedParam="
+                        + paramType[2]
+                        + "]"
+                );
+            }
+
+        }
```

Comment:
This section can be improved:
1. instead of using an array of bools, use a bitset (need only 3 bits) and appropriate masks (001, 010, ...) as you need to encode 4 states (1 unset, 3 different values).
2. instead of using checks on params1 and 2, add higher-level methods such as:
hasOnlyAnonymous() (which checks if any other bit than the target one is set), hasOnlyNamed..., etc...
3. The parsing exception is not user friendly - there's no need to specify all states rather the first that show a conflict along with example values namely the last param set (of the previous type) and the current param and its conflicting type, e.g. `"Inconsistent parameter declaration; use either anonymous or named params but not both: [1] or [{"name":2}]"`